### ISSUE TITLE
[cmd/telemetrygen] add support for profiles

### ISSUE THREAD

**Issue by haoqixu** [open-telemetry/opentelemetry-collector-contrib#36127]
### Component(s)

cmd/telemetrygen

### Describe the issue you're reporting

Add support for profiles to `telemetrygen` to support profiles in e2e tests.

related: #35999

**Comment by mx-psi** [comment]
@dmathieu is there any way we can do this today?

**Comment by dmathieu** [comment]
With rather simple data to reduce the amount breaking changes?

**Comment by mx-psi** [comment]
Ideally, yes :)

**Comment by dmathieu** [comment]
Actually, as I'm looking at this, it appears to be tricky to do.
There is no OTLP exporter for profiles, because the only location that does this export is the eBPF agent (the otel-go SDK doesn't handle profiles, and isn't meant to).

**Comment by fandreuz** [comment]
> Actually, as I'm looking at this, it appears to be tricky to do. There is no OTLP exporter for profiles, because the only location that does this export is the eBPF agent (the otel-go SDK doesn't handle profiles, and isn't meant to).

Hi @dmathieu, why is the otel-go SDK not meant to handle profiles? Is it documented somewhere?

**Comment by felixge** [comment]
Yeah, I'm surprised by this as well. I would expect the otel-go SDK to implement support for runtime profiling in the near future. The Go runtime offers lots of great profiling data and makes it easily accessible. I just needs to be converted to OTLP.

**Comment by dmathieu** [comment]
I'm not saying there can't be something that uses `runtime/pprof` to generate profiles and emit them.
But that there's no point in having a public API to generate profiles, like we have for logs, traces and metrics.

If we start consuming runtime data, the API should be very small (read cpu profile IO, export it somewhere).
One wouldn't manually generate profiles and add samples to them.
Hence the complexity in having telemetry generated for tests.

Similarly, the point of a public API is to be similar in all languages. This would be move like an instrumentation (like we have `otelhttp`, and every language has their owns).
They work similarly because they emit the same kind of data for the same behavior. But they're not explicitly documented, as they heavily depend on how the language works and its support for profiles.

> Is it documented somewhere in the spec?

No, there has been zero work on a public API in the spec.
